### PR TITLE
Update uplink set intial commit

### PR DESCRIPTION
**Pull request by nabhajit-ray**
### Description
[Fixed the update function of LIG so that SNMP, Switch Map, interconnect settings,and internal networks are does not get modified unnecessarily during LIG Update.]

### Issues Resolved
[#467 ]

### Check List
- [ ] New functionality includes testing.
  - [ ]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/resource_logical_interconnect_group.go [hunk 1]
```diff
@@ -1932,17 +2296,30 @@ func resourceLogicalInterconnectGroupRead(d *schema.ResourceData, meta interface
 	for i, snmpAccessIP := range logicalInterconnectGroup.SnmpConfiguration.SnmpAccess {
 		snmpAccess[i] = snmpAccessIP
 	}
+	// snmpUsers := make([]interface{}, len(logicalInterconnectGroup.SnmpConfiguration.SnmpUsers))
+	// for _, snmpUsersIP := range logicalInterconnectGroup.SnmpConfiguration.SnmpUsers {
```
Comment (AvijitOS) [inline oneview/resource_logical_interconnect_group.go:2300]: If it is not required, removed these commented code

File: oneview/resource_logical_interconnect_group.go [hunk 2]
```diff
@@ -2033,11 +2410,15 @@ func GetBoolPointer(value bool) *bool {
 
 func resourceLogicalInterconnectGroupUpdate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
+	//lig1, _ := config.ovClient.GetLogicalInterconnectGroupByName(d.Id())
```
Comment (AvijitOS) [inline oneview/resource_logical_interconnect_group.go:2413]: remove this commented line

File: oneview/resource_logical_interconnect_group.go [hunk 3]
```diff
@@ -1796,10 +2125,45 @@ func resourceLogicalInterconnectGroupRead(d *schema.ResourceData, meta interface
...
+	}
+
+	//Oneview send the uplink set in unordered way so ordering it.
+	matchedUplinkSets := make([]map[string]interface{}, 0)
+	unmatchedUplinkSets := make([]map[string]interface{}, 0)
+
+	if val, ok := d.GetOk("uplink_set"); ok {
+		oneviewuplinksetCount := len(uplinkSets)
+
+		uss := val.([]interface{})
```
Comment (AvijitOS) [inline oneview/resource_logical_interconnect_group.go:2139]: can u give some proper naming...something like uplinkSetsFromTfConfig

File: oneview/resource_logical_interconnect_group.go [hunk 4]
```diff
@@ -1796,10 +2125,45 @@ func resourceLogicalInterconnectGroupRead(d *schema.ResourceData, meta interface
...
+		uplinksetCount := len(uss)
+		for i := 0; i < uplinksetCount; i++ {
+			currName := (uss[i].(map[string]interface{}))["name"]
+			for j := 0; j < oneviewuplinksetCount; j++ {
+
+				if currName == uplinkSets[j]["name"] {
```
Comment (AvijitOS) [inline oneview/resource_logical_interconnect_group.go:2145]: i think its better
if uplinkSets[j]!= Nil and currName == uplinkSets[j]["name"]

Also, is OV uplinkset case insensitive? in that case better to make both of them to same case before comparison

File: oneview/resource_logical_interconnect_group.go (hunk 1 above)
Comment (nabhajit-ray) [reply oneview/resource_logical_interconnect_group.go:2301]: Done

File: oneview/resource_logical_interconnect_group.go (hunk 2 above)
Comment (nabhajit-ray) [reply oneview/resource_logical_interconnect_group.go]: Done

File: oneview/resource_logical_interconnect_group.go (hunk 3 above)
Comment (nabhajit-ray) [reply oneview/resource_logical_interconnect_group.go]: Done

File: oneview/resource_logical_interconnect_group.go (hunk 4 above)
Comment (nabhajit-ray) [reply oneview/resource_logical_interconnect_group.go]: Done

**Review by AvijitOS — APPROVED** [review]

**Review by alisha-k-kalladassery — APPROVED** [review]

**Pull request merged**